featurebyte/service/feature_manager.py

```diff
@@ -202,46 +227,157 @@ async def _populate_feature_store(
         schedule_time: datetime,
         aggregation_result_name: str,
     ) -> None:
-        next_job_time = get_next_job_datetime(
+        job_schedule_ts = get_previous_job_datetime(
             input_dt=schedule_time,
             frequency_minutes=tile_spec.frequency_minute,
             time_modulo_frequency_seconds=tile_spec.time_modulo_frequency_second,
         )
-        job_schedule_ts = next_job_time - timedelta(minutes=tile_spec.frequency_minute)
         job_schedule_ts_str = job_schedule_ts.strftime("%Y-%m-%d %H:%M:%S")
         await self.tile_manager_service.populate_feature_store(
             session, tile_spec, job_schedule_ts_str, aggregation_result_name
         )
 
-    async def _generate_historical_tiles(self, session: BaseSession, tile_spec: TileSpec) -> None:
-        # generate historical tile_values
-        date_format = "%Y-%m-%dT%H:%M:%S.%fZ"
+    async def _backfill_tiles(
+        self, session: BaseSession, tile_spec: TileSpec, schedule_time: datetime
+    ) -> None:
+        """
+        Backfill tiles required to populate internal online store
```

Comment:
This is the main change. Most of the other lines changed in the PR are tests to cover different scenarios and fixtures for checking executing queries.